services/products/index.js

```diff
@@ -84,6 +90,24 @@ const resolvers = {
       };
     },
   },
+  Subscription: {
+    productUpdate: {
+      subscribe: async function* () {
+        for(count = 0; count < 20; count++){
+          let product = products[Math.floor(Math.random()*products.length)];
+          let newProduct = {
+            upc: product.upc,
+            name: product.name,
+            price: Math.floor(Math.random() * 2000),
+            weight: Math.floor(Math.random() * 1000)
+          }
+
+          yield { productUpdate: newProduct };
+          await new Promise((resolve) => setTimeout(resolve, 3000));
```

Comment:
this can be done a little nicer with:
```js
const { setTimeout } = require("node:timers/promises");
```
```js
          await setTimeout(3000);
```

Reply:
I learned javascript more than 10 years ago and it shows 😂

Reply:
fixed in [900638e](https://github.com/apollographql/starstuff/pull/10/commits/900638e2dc49aef65008e06966289a617aa024cc)